### DISCUSSION TITLE
Cannot turn the drone to face inwards or outwards during a circle_left() or circle_right() maneuver

### DISCUSSION THREAD

**Discussion by 787-10** [bitcraze/discussions#835] [Q&A]
Hello Developer Team,

I'm currently interested in using the circle_left() and circle_right() functions, but with a modification in which the front of the drone is facing either inwards towards the center or outwards away from the center of the circular path. Currently, the observed behavior is for the forward of the drone to fly tangent with respect to the circular path which I believe is correct according to your implementation.

For example, in the start_circle_right() function

```
def start_circle_right(self, radius_m, velocity=VELOCITY):
        """
        Start a circular motion to the right. This function returns immediately

        :param radius_m: The radius of the circle (meters)
        :param velocity: The velocity of the motion (meters/second)
        :return:
        """
        circumference = 2 * radius_m * math.pi
        rate = 360.0 * velocity / circumference

        self._set_vel_setpoint(velocity, 0.0, 0.0, rate)
```

and the circle_right() function

```
def circle_right(self, radius_m, velocity=VELOCITY, angle_degrees=360.0):
        """
        Go in circle, clock wise

        :param radius_m: The radius of the circle (meters)
        :param velocity: The velocity along the circle (meters/second)
        :param angle_degrees: How far to go in the circle (degrees)
        :return:
        """
        distance = 2 * radius_m * math.pi * angle_degrees / 360.0
        flight_time = distance / velocity

        self.start_circle_right(radius_m, velocity)
        time.sleep(flight_time)
        self.stop()
```

I've been trying to adjust the parameters that are 0.0 inside the self.set_vel_setpoint() function call within the start_circle_right() function, but so far, I have not been able to produce the desired behavior. Any tips and assistance on this matter would be greatly appreciated.

**Comment by knmcguire** [comment]
Hi! I just transferred this issue to a discussion, as there we handle support questions. Issues indicate pieces of work that needs fixing in the code.

How about just changing the velocity input to affect the sideways velocity instead of the forward velocity? Like this:
`self._set_vel_setpoint(0.0, velocity, 0.0, rate)
`

I would imagine this producing exactly the result you are looking for, and that the rate affects if it it is looking outwards or inwards.

You can see the definition of set_vel_setpoint function here: 
https://github.com/bitcraze/crazyflie-lib-python/blob/f341d63c5e76dc7b35cf117a69da314de8083f87/cflib/positioning/motion_commander.py#L441

**Reply by 787-10** [reply]
Hi, thank you very much for your response. I am new to the discussions feature, and will be making sure to use this going forward - sorry for the inconvenience caused with this originally being in issues.

Your suggestion worked - I was able to get the drone to fly in a circle with the front end pointing inwards radially.

A couple of follow up questions:
(1) If I want to get the drone to fly in a circle with the front end pointing outwards radially, what would the changed parameters look like? Would it be something like `self._set_vel_setpoint(0.0, -velocity, 0.0, rate) ` or something else?

(2) It seems that for the current implementation of the function, the drone will be flying in a circle using the starting point where the function is called as a center. Let's say that I want to change the location of the center from time to time by giving the drone a set of new coordinates for it to fly a circle around. Would it be possible to build that into the current implementation -- to have the drone fly in a circle around a center that is changing positions every few seconds or so? Any tips regarding this would be very helpful, thanks in advance!

**Comment by krichardsson** [comment]
> (1) If I want to get the drone to fly in a circle with the front end pointing outwards radially, what would the changed parameters look like? Would it be something like self._set_vel_setpoint(0.0, -velocity, 0.0, rate)  or something else?

Looks good, did you try it?

> (2) It seems that for the current implementation of the function, the drone will be flying in a circle using the starting point where the function is called as a center. Let's say that I want to change the location of the center from time to time by giving the drone a set of new coordinates for it to fly a circle around. Would it be possible to build that into the current implementation -- to have the drone fly in a circle around a center that is changing positions every few seconds or so?

You can move to other locations using the various commands such as `forward()`, `back()` and so on, [check out the API for options](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/positioning/motion_commander/).
